eventdisplay/FairXMLNode.cxx

```diff
@@ -8,232 +8,105 @@
-		return FairXMLNode::NullInstance();
-	}
 	Int_t control_index = 0;
-	for(int i=0;i<fChildren->GetEntries();i++){
-		FairXMLNode *node = (FairXMLNode*)fChildren->At(i);
+	for(int i=0;i<fChildren.GetEntries();i++){
+		FairXMLNode *node = GetChild(i);
 		TString temp = node->GetName();
 		if(temp==name){
 			control_index++;
 		}
 		if(control_index>count)
 			return node;
 	}
-	//FairCout::Error("No child found by name");
-	return FairXMLNode::NullInstance();
-}
-
-FairXMLNode* FairXMLNode::GetChild(Int_t index) const{
-	if(fMode == kModeNull){
-		return FairXMLNode::NullInstance();
-	}
-	if(fNChildren>index){
-		return (FairXMLNode*)fChildren->At(index);
-	}else{
-		return FairXMLNode::NullInstance();
-	}
-}
-
-Bool_t FairXMLNode::IsImport() const{
-	if(fMode==kModeImport) return kTRUE;
-	return kFALSE;
-}
-
-Bool_t FairXMLNode::IsNull() const {
-	if(fMode == kModeNull) return kTRUE;
-	return kFALSE;
-}
-
-FairXMLNode* FairXMLNode::NullInstance() {
-	if(fgNullInstance){
-		return fgNullInstance;
-	}
-	fgNullInstance = new FairXMLNode();
-	return fgNullInstance;
-}
-
-FairXMLNode::FairXMLNode() {
-	fName = "null";
-	fValue= "null";
-	fChildren = NULL;
-	fAttrib = NULL;
-	fNChildren = 0;
-	fNAttrib = 0;
-	fMode = kModeNull;
+	return nullptr;
 }
 
-void FairXMLNode::SetValue(TString value) {
-	if(fMode==kModeExport){
-		fValue = value;
+FairXMLAttrib* FairXMLNode::GetAttrib(TString name) const{
+	Int_t control_index = 0;
+	for(int i=0;i<fAttrib.GetEntries();i++){
+		FairXMLAttrib *node = static_cast<FairXMLAttrib*>(fAttrib.At(i));
+		TString temp = node->GetName();
+		if(temp==name){
+			return node;
+		}
 	}
+	return nullptr;
 }
 
-void FairXMLNode::SetName(TString name) {
-	if(fMode==kModeExport){
-		fName = name;
+FairXMLNode* FairXMLNode::GetChild(Int_t index) const{
+	if(index<fChildren.GetEntries()){
+		return static_cast<FairXMLNode*>(fChildren.At(index));
 	}else{
+		return nullptr;
 	}
 }
 
-TString FairXMLNode::GetValue() const {
-	if(fMode==kModeNull){
-		return "";
-	}
-	return fValue;
-}
-
-const char* FairXMLNode::GetName() {
-	if(fMode==kModeNull){
-		return "";
-	}
-	return fName.Data();
-}
-
-Bool_t FairXMLNode::IsExport() const{
-	if(fMode==kModeNull){
-		return kTRUE;
-	}
-	return kFALSE;
-}
-
-TString FairXMLNode::GetAttribValue(TString name, Int_t index) const{
-	if(fMode==kModeNull){
-		return "";
-	}
-	if(fNAttrib==0){
-		return "";
-	}
-	int counter =-1;
-	for(int i=0;i<fAttrib->GetEntries();i++){
-		FairXMLAttrib *atrib = (FairXMLAttrib*)fAttrib->At(i);
-		TString title = atrib->GetName();
-		if(title == name){
-			counter++;
-		}
-		if(counter==index){
-			return atrib->GetValue();
+void FairXMLNode::AddAttrib(FairXMLAttrib* attrib) {
+	TString new_atr = attrib->GetName();
+	for(int i=0;i<GetNAttributes();i++){
+		TString atr = GetAttrib(i)->GetName();
+		if(atr.EqualTo(new_atr)){
+			std::cout<<"FairXMLNode::AddAttrib Can't have two attributes with the same name!"<<std::endl;
+			return;
 		}
```

Comment:
This can probably also be simplified with one of [`TList::FindObject`](https://root.cern.ch/doc/master/classTList.html#a85c8734ab131c0db5f68c724e47496a7) overloads.